test/CapabilitiesTest.php

```diff
@@ -98,6 +98,52 @@ public function testSilencedDeprecation(): void
         self::assertTrue(true);
     }
 
+    private function triggerWarning(): void
+    {
+        foreach (1 as $n) {}
+    }
+
+    public function testDuplicateWarningsA(): void
+    {
+        $this->triggerWarning();
+
+        self::assertTrue(true);
+    }
+
+    public function testDuplicateWarningsB(): void
+    {
+        $this->triggerWarning();
+        $this->triggerWarning();
+        $this->triggerWarning();
+
+        foreach (1 as $n) {}
+
+        self::assertTrue(true);
+    }
+
+    public function testDuplicateWarningsC(): void
+    {
+        self::assertTrue(true);
+    }
+
+    public function testMixedSeverities(): void
+    {
+        // Silenced warning.
+        $foo = @$bar;
+
+        // Notice.
+        $foo = &self::provideData();
+        // Warning.
+        foreach (1 as $n) {}
+        // Deprecated.
+        trim(null);
+
+        // Silenced warning.
+        $foo = @$bar;
```

Comment:
Is it necessary to include silencing here? Isn't that somewhat unrelated to this test and covered elsewhere?

Reply:
Makes sense.

Reply:
I've removed it, just to discover it was masking a formatting bug.

I'm having a ridiculously hard time trying to get a consistent and predictable number of carriage returns around issue messages. Tests expect that ANSI colour codes surround new line characters, and I didn't want to rewrite all tests, so I tried to keep existing behaviour. I never get it right, there's always a missed case.

I'll try again tomorrow, after some sleep.